### PR TITLE
Fixes #23 iocage package is deprecated

### LINKED ISSUE #23: iocage package is deprecated
iocage is dead, long live py-iocage

we should change the install package to `py27-iocage`, as well as the binary [referenced in the provider](https://github.com/xaque208/puppet-jail/blob/master/lib/puppet/provider/jail/iocage.rb#L8)

the service name remains unchanged.

### PR DESCRIPTION
**Pull request by xaque208**
Here we replace the package name of iocage, while leaving it tunable for
environments that do not use the default python version on FreeBSD.  Also here,
the bin path is adjusted.

### CODE REVIEW

**Comment by xaque208** [comment]
@igalic If you have time to test this, I'd appreciate it.

File: lib/puppet/provider/jail/iocage.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
   confine    kernel: :freebsd
   defaultfor kernel: :freebsd
 
-  commands iocage: '/usr/local/sbin/iocage'
+  commands iocage: '/usr/local/bin/iocage'
```
Comment (igalic) [inline lib/puppet/provider/jail/iocage.rb]: imo, i'd change this to simply `'iocage'`

File: metadata.json [hunk 2]
```diff
@@ -23,6 +23,6 @@
     }
   ],
   "dependencies": [
-    { "name": "puppetlabs/concat", "version_requirement": ">= 1.1.0" }
+    { "name": "puppetlabs/concat", "version_requirement": ">= 1.1.0 < 3.0.0" }
```
Comment (igalic) [inline metadata.json]: this is not `stdlib` as the commit message suggests.

Comment (xaque208) [inline metadata.json]: The commit message has been corrected.

**Comment by xaque208** [comment]
I've added another commit to this PR to ensure that `puppet resource jail` still works on my machines.  It certainly seems slower than the previous version.

**Comment by xaque208** [comment]
`time iocage get all $jail` takes 5 seconds on the new py-iocage.  The same command on the old shell version takes 0.5 seconds.

File: lib/puppet/provider/jail/iocage.rb [hunk 3]
```diff
@@ -5,21 +5,34 @@
...
 
-  commands iocage: '/usr/local/sbin/iocage'
+  commands iocage: 'iocage'
 
   mk_resource_methods
 
   def self.jail_list
-    output = iocage(['list']).split("\n")
-    fields = output.shift.split.map { |i| i.downcase.to_sym }
+    output = execute('iocage list -l', override_locale: false).split("\n")
```
Comment (igalic) [inline lib/puppet/provider/jail/iocage.rb]: we could simplify this whole block by adding `-H` to the command line

**Pull request closed without merging**